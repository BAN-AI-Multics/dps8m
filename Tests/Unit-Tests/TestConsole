set debug stdout
do ../../src/dps8/base_system.ini
reset scu
;set iom debug=all
;set opcon debug=all
;set cpu debug=trace;intr;fault
load TestConsole.oct
g 0
q

