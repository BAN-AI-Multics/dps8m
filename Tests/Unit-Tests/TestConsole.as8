	bool	connect_channel, 2
	bool	console_channel, 36

        bool	mailbox_base_address, 01400
        bool	connect_mbx, 01410	" base + 4 * chan_num
	bool	console_mbx, 01570

	bool	CONSOLE.write_ASCII, 033
	bool	CONSOLE.write_alert, 051
	bool	console_dev_code, 0

	org	0
	tra	test

	org	connect_mbx
connect_lpw:
	oct	0		" LPW
	oct	0		" LPWE
	oct	0		" SCW
	oct	0		" DCW



	org	010000
test:

" Print "Console test"

	emcall	8
	"absa	msg1
	eaa	msg1
	emcall	16
	emcall	8

" Print "Expect ALERT"

	eaa	msg2
	emcall	16
	emcall	8

" Send an ALERT

	lda	lpw1
	sta	connect_lpw
	cioc	cioc_port

" Print 'Expect "Hello, world"'

"	emcall	8
"	eaa	msg3
"	emcall	16
"	emcall	8
"
"	lda	lpw2
"	sta	connect_lpw
"	cioc	cioc_port



	dis

" Data

cioc_port:
	dec	0

	even

	bool	non_data,2
	bool	terminate,0

" ALERT

lpw1:	vfd	18/pwc1,6/2,12/1

	even
pwc1:
	vfd	6/CONSOLE.write_alert,6/console_dev_code,6/0,3/7,1/0,2/terminate,6/non_data,6/0
	vfd	9/console_channel,27/0

" write.ASCII


lpw2:	vfd	18/pcw2,6/2,12/1

	even
pcw2:
	vfd	6/CONSOLE.write_ASCII,6/console_dev_code,6/0,3/7,1/0,2/terminate,6/non_data,6/0
	vfd	9/console_channel,27/0

msg1:	dec	12
	aci	'Console test'

msg2:	dec	12
	aci	'Expect ALERT'

msg3:	dec	20
	aci	'Expect "Hello, word"'

