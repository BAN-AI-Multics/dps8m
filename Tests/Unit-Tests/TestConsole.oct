!SIZE 010046
000000 xxxx 010000710000 	tra	test
000030 xxxx 000000011000 	nop
000031 xxxx 000000011000 	nop
001410 xxxx 000000000000 	oct	0		" LPW
001411 xxxx 000000000000 	oct	0		" LPWE
001412 xxxx 000000000000 	oct	0		" SCW
001413 xxxx 000000000000 	oct	0		" DCW
010000 xxxx 000010420400 	emcall	8
010001 xxxx 010030635000 	eaa	msg1
010002 xxxx 000020420400 	emcall	16
010003 xxxx 010034635000 	eaa	msg2
010004 xxxx 000020420400 	emcall	16
010005 xxxx 010020235000 	lda	lpw1
010006 xxxx 001410755000 	sta	connect_lpw
010007 xxxx 010017015000 	cioc	cioc_port
010010 xxxx 000010420400 	emcall	8
010011 xxxx 010040635000 	eaa	msg3
010012 xxxx 000020420400 	emcall	16
010013 xxxx 010024235000 	lda	lpw2
010014 xxxx 001410755000 	sta	connect_lpw
010015 xxxx 010017015000 	cioc	cioc_port
010016 xxxx 000022420400 	emcall	18
010017 xxxx 000000000000 	dec	0
010020 xxxx 010022020001 lpw1:	vfd	18/pwc1,6/2,12/1
010021 xxxx 000000011000 	even "(allocating 1 nop)
010022 xxxx 510000700200 	vfd	6/CONSOLE.write_alert,6/console_dev_code,6/0,3/7,1/0,2/terminate,6/non_data,6/0
010023 xxxx 036000000000 	vfd	9/console_channel,27/0
010024 xxxx 010026020001 lpw2:	vfd	18/pcw2,6/2,12/1
010025 xxxx 000000011000 	even "(allocating 1 nop)
010026 xxxx 000000700200 	vfd	6/CONSOLE.request_status,6/console_dev_code,6/0,3/7,1/0,2/terminate,6/non_data,6/0
010027 xxxx 036000000000 	vfd	9/console_channel,27/0
010030 xxxx 103157156163 msg1:	aci	'Console test\n\0'
010031 xxxx 157154145040 
010032 xxxx 164145163164 
010033 xxxx 012000000000 
010034 xxxx 105170160145 msg2:	aci	'Expect ALERT\n\0'
010035 xxxx 143164040101 
010036 xxxx 114105122124 
010037 xxxx 012000000000 
010040 xxxx 105170160145 msg3:	aci	'Expect "Hello, world"\n\0'
010041 xxxx 143164040042 
010042 xxxx 110145154154 
010043 xxxx 157054040167 
010044 xxxx 157162154144 
010045 xxxx 042012000000 
