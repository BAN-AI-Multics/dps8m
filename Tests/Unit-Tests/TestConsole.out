57 (000071) words loaded

dps-8/m simulator V4.0-0 Beta        git commit id: c8ae76cb
Debug output to "STDOUT"
Debug output to "STDOUT" at Wed Feb 19 15:37:55 2014

Console test
Expect ALERT
CONSOLE: ALERT

leaving DIS_cycle

Expect "Hello, world"
CONSOLE: Hello, world

cpuCycles = 21
TestConsole-8> g 0

Halt, IC: 010020 (000022420400)
Goodbye
Debug output disabled
